src/software/ai/hl/stp/stp_tactic_assignment_test.cpp

```diff
@@ -440,16 +440,21 @@ TEST_F(STPTacticAssignmentTest,
 
     stp.assignRobotsToTactics(world, tactics);
 
+    EXPECT_TRUE(allTacticsAssigned(tactics));
     ASSERT_TRUE(goalie_tactic_1->getAssignedRobot().has_value());
     EXPECT_EQ(goalie_tactic_1->getAssignedRobot().value(), robot_0);
 
     // Change the goalie and perform the same check in case we have a fluke bug
+    // Re-insert goalie tactic into queue since goalie tactics are removed after
+    // assignment
+    tactics.insert(tactics.begin(), goalie_tactic_1);
```

Comment:
Why do we need to add this now, but we didn't need to before?

Reply:
Because previously, `tactics` was not returned or mutated  So although we assigned the goalie tactic (which is removed from `tactics` in the assign function), goalie is still in there and we could just re-use it for the 2nd test.
Now that we return `tactics`, we have to re-insert it or else, the goalie tactic wouldn't be in the `tactics` queue.

Reply:
Removed default arg.